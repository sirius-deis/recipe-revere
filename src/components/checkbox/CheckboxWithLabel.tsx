import { FC, useId } from "react";
import styles from "./CheckboxWithLabel.module.css";

interface CheckboxWithLabelProps {
  isChecked: boolean;
  labelText: string;
  onCheckHandler: () => void;
}

const CheckboxWithLabel: FC<CheckboxWithLabelProps> = ({
  isChecked,
  labelText,
  onCheckHandler,
}) => {
  const id = useId();
  return (
    <div>
      <label className={styles.label}>
        {labelText}
        <label htmlFor={id} className={styles.container}>
          <input
            type="checkbox"
            id={id}
            checked={isChecked}
            onChange={onCheckHandler}
          />
          <span className={styles.checkmark}></span>
        </label>
      </label>
    </div>
  );
};

export default CheckboxWithLabel;
