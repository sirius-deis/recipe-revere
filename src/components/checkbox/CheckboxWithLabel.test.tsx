import { render, screen, fireEvent } from "@testing-library/react";
import CheckboxWithLabel from "./CheckboxWithLabel";

describe("CheckboxWithLabel component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <CheckboxWithLabel
        isChecked={false}
        labelText="Remember me"
        onCheckHandler={() => {}}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it("should render component with proper text", () => {
    render(
      <CheckboxWithLabel
        isChecked={false}
        labelText="Remember me"
        onCheckHandler={() => {}}
      />
    );
    expect(screen.getByText("Remember me")).toBeInTheDocument();
  });
  it("should call the function when a change event is fired", () => {
    const onCheckHandler = jest.fn();
    render(
      <CheckboxWithLabel
        isChecked={false}
        labelText="Remember me"
        onCheckHandler={onCheckHandler}
      />
    );
    const checkboxLabel = screen.getByLabelText("Remember me");
    fireEvent.click(checkboxLabel);
    expect(onCheckHandler).toHaveBeenCalled();
    expect(onCheckHandler).toHaveBeenCalledTimes(1);
    const checkbox = checkboxLabel?.nextElementSibling;
    fireEvent.click(checkbox!);
    expect(onCheckHandler).toHaveBeenCalledTimes(2);
  });
});
