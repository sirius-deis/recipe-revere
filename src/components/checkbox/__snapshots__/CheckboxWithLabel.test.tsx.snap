// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxWithLabel component should match snapshot 1`] = `
<div>
  <label
    class="checkbox"
  >
    <label
      class="container"
      for=":r0:"
    >
      <input
        id=":r0:"
        type="checkbox"
      />
      <span
        class="checkmark"
      />
    </label>
    Remember me
  </label>
</div>
`;
